libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -78,30 +80,38 @@ def _convert_oper_to_sql(oper: str) -> str:
     return oper_map.get(oper, "==")
 
 
-def _generate_condition(condition: FilterCondition) -> str:
-    key = condition["key"]
+def _generate_condition(condition: FilterCondition, bind_variables: List[str]) -> str:
+    key = _quote_indentifier(condition["key"])
     oper = _convert_oper_to_sql(condition["oper"])
     value = condition["value"]
-    if isinstance(value, str):
-        value = f'"{value}"'
-    return f"JSON_EXISTS(metadata, '$.{key}?(@ {oper} {value})')"
+
+    value_bind = f":value{len(bind_variables)}"
+    bind_variables.append(value)
+
+    return (
+        f"JSON_EXISTS(metadata, "
+        f"'$.{key}[*]?(@ {oper} $val)' "
```

Comment:
Can you confirm on [*]
[*] means it checks contains right? But we are checking for key = value or key > value but not key contains value .. 


like metadata has `"tags": ["db", "oracle", "json"]`, 
if user gives key: tags,  oper:EQ , value : oracle , It would work which should have failed because it contains other values db and json..

Reply:
Again, talking for the new format, equality is meant to be used with scalars. If size known users can use indexing {"tags[0]" : "db", "tags[1]" : "oracle", "tags[2]" : "json"}. Maybe we can add a $size operator to make sure they can implement equality with lists.

Reply:
I added list and dict equality using JSON_QUERY and JSON_EQUAL in the last commit.